added support mapper 7  , 66  & partial support mapper 11

SuzieQQ opened the pull request:
hello I have added support for mapper 7

amhndu commented on src/MapperAxROM.cpp:
```diff
@@ -0,0 +1,91 @@
+
+            m_characterRAM.resize(0x2000);
+            m_ppubus.m_RAM.resize(0x400);
+            std::cout << "Uses Character RAM OK" << std::endl;
```
This could be a `Log(Info)` instead of cout

amhndu commented on src/MapperAxROM.cpp:
```diff
@@ -0,0 +1,91 @@
+
+            for (int i = 0; i < 0x400; i++)
+            {
+                m_ppubus.m_RAM[i] = 0;
```
I'm not sure what this is supposed to do here. `AxROM.h` creates a new ppubus that's not used by anything else?
Should this be a different array managed by the mapper or does this require something in the cartridge?

amhndu commented on src/MapperAxROM.cpp:
```diff
@@ -0,0 +1,91 @@
+        
+        return m_cartridge.getROM()[base + PRGBank * 0x8000 + (address & 0x7FFF)];
+
+        }
```
we need to handle `else` case

amhndu commented:
Thanks for the pull request! Good work!
I've tried it with battletoads and it works! I've had some issues with Aladdin. I haven't gone deep into how AxROM works yet, so I don't know exactly what's wrong. For now, I've left some comments that could be issues and will do a deeper review in a while.

amhndu commented on src/Mapper.cpp:
```diff
@@ -28,9 +29,11 @@ namespace sn
                 ret.reset(new MapperCNROM(cart));
                 break;
+            case AxROM:
+            ret.reset(new MapperAxROM(cart,mirroring_cb));
```
This should have a `break;` statement. It works now only because the default is empty but it could be an issue later

amhndu commented on src/Emulator.cpp:
```diff
@@ -61,7 +61,7 @@ namespace sn
 
         m_window.create(sf::VideoMode(NESVideoWidth * m_screenScale, NESVideoHeight * m_screenScale),
-                        "SimpleNES", sf::Style::Titlebar | sf::Style::Close);
+                        "SimpleNES", sf::Style::Titlebar | sf::Style::Close | sf::Style::Resize);
```
Adding resize also allows the user to set the window to incorrect aspect ratio. That's why I initially didn't allow it and instead supported a command line argument to resize. But this is okay for now.

SuzieQQ replied:
sorry i read only now, thanks :) i know that aladdin if i remember correctly uses illegal opcodes.

mayweather911 commented:
> Thanks for the pull request! Good work!
> I've tried it with battletoads and it works! I've had some issues with Aladdin. I haven't gone deep into how AxROM works yet, so I don't know exactly what's wrong. For now, I've left some comments that could be issues and will do a deeper review in a while.

Sorry I've only just read, hello thanks, I'm new and have been programming for almost 6 months! I honestly did what I could 

I know Aladdin uses illegal opcodes so maybe that's why it doesn't work

SuzieQQ commented:
> Thanks for the pull request! Good work! I've tried it with battletoads and it works! I've had some issues with Aladdin. I haven't gone deep into how AxROM works yet, so I don't know exactly what's wrong. For now, I've left some comments that could be issues and will do a deeper review in a while.



sorry i read only now, thanks :) i know that aladdin if i remember correctly uses illegal opcodes.

SuzieQQ commented:
> Thanks for the pull request! Good work! I've tried it with battletoads and it works! I've had some issues with Aladdin. I haven't gone deep into how AxROM works yet, so I don't know exactly what's wrong. For now, I've left some comments that could be issues and will do a deeper review in a while.

hi i updated i managed to have almost complete the mapper 66 all games work
but no doraemon crashes in the menu.

I added mapper 11 but for now only two Metal Fighter & Shockwave games work

SuzieQQ commented:
hey hello i'm trying to insert mapper 4 but it's still a bit missing. here are screenshots of some games


https://github.com/SuzieQQ/screennes/tree/main/TEST-MAPPER-4-GAME-SCREEN

amhndu commented:
Hey, sorry about the long delay (been busy with stuff). Let's first get these mappers merged. You haven't responded to my requested changes. Can you go through those, along with making sure the formatting of the files are correct (eg. GxROM has some weird indents)?
Meanwhile I'll re-read NES docs. I've forgotton all about it. 
For mapper, we can use #23 to discuss.

SuzieQQ commented:
here I am I updated the lemulator and formatted with ide the mapperGxROM aka 66

inserted the mapper 4

> Hey, sorry about the long delay (been busy with stuff). Let's first get these mappers merged. You haven't responded to my requested changes. Can you go through those, along with making sure the formatting of the files are correct eg. GxROM has some weird indents (try using your IDE auto-formatters to fix indentation)? Meanwhile I'll re-read NES docs. I've forgotton all about it.
> 
> For mapper 4, we can use #23 to discuss. Feel free to also email me directly.

here I am I updated the lemulator and formatted with ide the mapperGxROM aka 66

inserted the mapper 4
